staging/src/k8s.io/kubectl/pkg/polymorphichelpers/logsforobject.go

```diff
@@ -69,6 +73,16 @@ func logsForObjectWithClient(clientset corev1client.CoreV1Interface, object, opt
 		return ret, nil
 
 	case *corev1.Pod:
+		// in case the "kubectl.kubernetes.io/default-logs-container" annotation is present, we preset the opts.Containers to default to selected
+		// container. This gives users ability to preselect the most interesting container in pod.
+		if annotations := t.GetAnnotations(); annotations != nil && len(opts.Container) == 0 && len(annotations[defaultLogsContainerAnnotationName]) > 0 {
+			containerName := annotations[defaultLogsContainerAnnotationName]
+			if exists, _ := findContainerByName(t, containerName); exists != nil {
```

Comment:
Should we exclude init and ephemeral containers or there are some use-cases for them too?

Reply:
yeah, i would leave it to pod creator to define what he think is the most important container logs he want to see.